### ISSUE TITLE
Can't parse XHTML 1.0 document: Cannot resolve external DTD subset - public ID = '-//W3C//DTD XHTML 1.0 Strict//EN', system ID = 'http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd'

### ISSUE THREAD

**Issue by martin-honnen** [xmlresolver/sampleappcs#1]
I have tried to pass the following XHTML 1 document to the sample app:

```
<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html 
     PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
    "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">
<head>
    <title>Test</title>
</head>
<body>
    <h1>Test</h1>
    <p>This is a test for 0 &euro;</p>
</body>
</html>

```
I get an error:

```
Parsing C:\Users\Martin Honnen\Source\Repos\XmlResolverNETCoreTest1\XmlResolverNETCoreTest1\sample1.xhtml
Using catalogs: ./catalog.xml, pack://application:,,,XmlResolverData;0.1.0.0;component/Org.XmlResolver.catalog.xml
Well, that didn't work like we expected, did it?
Cannot resolve external DTD subset - public ID = '-//W3C//DTD XHTML 1.0 Strict//EN', system ID = 'http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd'.
```

**Comment by martin-honnen** [comment]
The `cat-xhtml.xml` contains an entry
```
<system systemId="https://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"
           uri="www_w3_org.TR.xhtml1.DTD.xhtml1-strict.dtd"/>
```

Shouldn't that be `http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd` (i.e. `http` instead of `https`)?

**Comment by martin-honnen** [comment]
I have tried the Java sample application in the meantime and it manages to parse and validate the XHTML file. And it looks as if the catalog systemIds there also use `https` www.w3.org URIs so my previous comment probably isn't the right suspicion why the CS version fails to resolve the XHTML 1 DTD.

**Comment by martin-honnen** [comment]
Stacktrace comes from .NET classes:
```
   at System.Xml.XmlTextReaderImpl.Throw(Exception e)
   at System.Xml.XmlTextReaderImpl.PushExternalEntityOrSubset(String publicId, String systemId, Uri baseUri, String entityName)
   at System.Xml.XmlTextReaderImpl.DtdParserProxy_PushExternalSubset(String systemId, String publicId)
   at System.Xml.XmlTextReaderImpl.DtdParserProxy.System.Xml.IDtdParserAdapter.PushExternalSubset(String systemId, String publicId)
   at System.Xml.DtdParser.ParseExternalSubset()
   at System.Xml.DtdParser.ParseInDocumentDtd(Boolean saveInternalSubset)
   at System.Xml.DtdParser.Parse(Boolean saveInternalSubset)
   at System.Xml.DtdParser.System.Xml.IDtdParser.ParseInternalDtd(IDtdParserAdapter adapter, Boolean saveInternalSubset)
   at System.Xml.XmlTextReaderImpl.ParseDtd()
   at System.Xml.XmlTextReaderImpl.ParseDoctypeDecl()
   at System.Xml.XmlTextReaderImpl.ParseDocumentContent()
   at System.Xml.XmlTextReaderImpl.Read()
   at System.Xml.XmlValidatingReaderImpl.Read()
   at SampleApp.Program.Main(String[] args) in C:\Users\Martin Honnen\Source\Repos\sampleappxmlresolvercs\SampleApp\SampleApp\Program.cs:line 53
```

If I pass the file to .NET's `XmlPreloadedResolver`, as in
```
            using (XmlReader xmlReader = XmlReader.Create("sample1.xhtml", new XmlReaderSettings { DtdProcessing = DtdProcessing.Parse, ValidationType = ValidationType.DTD, XmlResolver = new XmlPreloadedResolver(new XmlUrlResolver()) }))
            {
                while (xmlReader.Read())
                {
                    Console.WriteLine(xmlReader.NodeType);
                }
            }
```

all works fine, file is parsed and validated.

**Comment by martin-honnen** [comment]
If I use a local catalog with local copies of all files on the file system, like:
```
<?xml version="1.0" encoding="utf-8" ?>
<catalog xmlns="urn:oasis:names:tc:entity:xmlns:xml:catalog">
  <system systemId="http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"
          uri="dtds/xhtml1-strict.dtd"/>
  <system systemId="http://www.w3.org/TR/xhtml1/DTD/xhtml-lat1.ent"
          uri="dtds/xhtml-lat1.ent"/>
  <system systemId="http://www.w3.org/TR/xhtml1/DTD/xhtml-symbol.ent"
          uri="dtds/xhtml-symbol.ent"/>
  <system systemId="http://www.w3.org/TR/xhtml1/DTD/xhtml-special.ent"
          uri="dtds/xhtml-special.ent"/>
</catalog>

```

and the `-c` option of the sample app, the document is parsed fine.

**Comment by ndw** [comment]
This was a bug in the data assembly. Thank you, Martin! Fixed in the 0.1.1 release of XmlResolverData.
